pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -316,6 +638,16 @@ func (a *sqlInstanceAdapter) Update(ctx context.Context, updateOp *directbase.Up
 		a.actual = updated
 	}
 
+	isEditionUnmanaged := false
```

Comment:
Do we still need this?  I would expect if the edition is unmanaged that the values should match

Reply:
I think we still need to do a bit of extra work as `edition` needs to be patched separately from other fields.

Reply:
Right, I'm saying shouldn't the values match if the edition is unmanaged?  That's how it works for other fields, IIUC